Kubernetes: Correct artifact path location

Sgitario opened the pull request:
Adding dependencies like `quarkus-openshift` and `quarkus-kubernetes-config` might lead to running the Kubernetes processor when the JAR file is not correct yet. 

With these changes, we'll get always the correct artifact. 
Fix https://github.com/quarkusio/quarkus/issues/28108

Sgitario commented:
cc @iocanel @michalvavrik

I've tested these changes in JVM and native, but I'd appreciate if you have another round to ensure that it works the same for Native before merging @michalvavrik.

michalvavrik commented:
Hello @Sgitario , unfortunately we are having cluster issues today, but I check once I can.

michalvavrik commented:
Thank you for the fix btw.

michalvavrik commented:
I run tests - reproducer in JVM and Quarkus QE TS in JVM/native are fixed. Reproduced failed in native mode, but it also failed when I run it with current main (exception looks not related).

Sgitario commented:
It seems that the current solution introduces build cycle issues: `GeneratedResourceBuildItem` -> `JarBuildItem` - > KubernetesProcessor... and again
So, I will need to find another solution.

Sgitario committed:
Kubernetes: Correct artifact path location

Adding dependencies like `quarkus-openshift` and `quarkus-kubernetes-config` might lead into running the kubernetes processor when the JAR file is not correct yet. 

With these changes, we'll get always the correct artifact.
Fix https://github.com/quarkusio/quarkus/issues/28108

geoand commented:
Is this now ready to go @Sgitario ?

Sgitario commented:
> Is this now ready to go @Sgitario ?

From my point of view, yes +1

iocanel reviewed (approved):
LGTM
